`NoSuchElementException: No value present` exception at sync completion

After completing sync process

```
2025-07-08 13:18:24.330 INFO  - Syncing     *** Slot: 87392, Head slot: 87296, Remaining slots: 96, Target chain: 92ff5b..c638 (87372) with 16 peers, Connected peers: 54
2025-07-08 13:18:24.330 INFO  - Sync Info   *** Range: 87297 - 87372 (4 batches), Downloading: 25 slots (1 batches), Ready: 51 slots (3 batches), Batch import: waiting
2025-07-08 13:18:28.064 INFO  - Starting eth2 gossip
2025-07-08 13:18:28.085 INFO  - Using custody sidecar subnets count: 128
2025-07-08 13:18:28.095 INFO  - Starting gossip for BPO fork (fork digest: 0xb62f2b0e) scheduled at epoch 1584 with 20 max blobs per block
2025-07-08 13:18:28.134 INFO  - Subscribe to topic: /eth2/b62f2b0e/beacon_block/ssz_snappy
2025-07-08 13:18:36.002 INFO  - Syncing     *** Slot: 87393, Head slot: 87296, Remaining slots: 97, Target chain: 92ff5b..c638 (87372) with 16 peers, Connected peers: 55
2025-07-08 13:18:36.002 INFO  - Sync Info   *** Range: 87297 - 87372 (4 batches), Downloading: none, Ready: 76 slots (4 batches), Batch import: in progress
2025-07-08 13:18:48.001 INFO  - Syncing     *** Slot: 87394, Head slot: 87297, Remaining slots: 97, Target chain: 92ff5b..c638 (87372) with 16 peers, Connected peers: 56
2025-07-08 13:18:48.002 INFO  - Sync Info   *** Range: 87297 - 87372 (4 batches), Downloading: none, Ready: 76 slots (4 batches), Batch import: in progress
2025-07-08 13:19:00.002 INFO  - Syncing     *** Slot: 87395, Head slot: 87314, Remaining slots: 81, Target chain: 92ff5b..c638 (87372) with 16 peers, Connected peers: 56
2025-07-08 13:19:00.002 INFO  - Sync Info   *** Range: 87297 - 87372 (4 batches), Downloading: none, Ready: 76 slots (4 batches), Batch import: in progress
2025-07-08 13:19:12.003 INFO  - Syncing     *** Slot: 87396, Head slot: 87356, Remaining slots: 40, Target chain: 92ff5b..c638 (87372) with 16 peers, Connected peers: 56
2025-07-08 13:19:12.004 INFO  - Sync Info   *** Range: 87347 - 87372 (2 batches), Downloading: none, Ready: 26 slots (2 batches), Batch import: in progress
2025-07-08 13:19:17.795 INFO  - Beacon chain syncing complete, waiting for Execution Client
2025-07-08 13:19:24.011 INFO  - Syncing     *** Slot: 87397, Head slot: 87372, Waiting for execution layer sync, Connected peers: 57
2025-07-08 13:19:35.836 INFO  - Syncing completed
2025-07-08 13:19:35.837 INFO  - Starting gossip for BPO fork (fork digest: 0xb62f2b0e) scheduled at epoch 1584 with 20 max blobs per block
2025-07-08 13:19:35.840 INFO  - Subscribe to topic: /eth2/b62f2b0e/beacon_attestation_25/ssz_snappy
2025-07-08 13:19:35.840 INFO  - Subscribe to topic: /eth2/b62f2b0e/beacon_attestation_26/ssz_snappy
2025-07-08 13:19:35.840 INFO  - Subscribe to topic: /eth2/b62f2b0e/beacon_aggregate_and_proof/ssz_snappy
2025-07-08 13:19:35.840 INFO  - Subscribe to topic: /eth2/b62f2b0e/voluntary_exit/ssz_snappy
2025-07-08 13:19:35.841 INFO  - Subscribe to topic: /eth2/b62f2b0e/proposer_slashing/ssz_snappy
2025-07-08 13:19:35.841 INFO  - Subscribe to topic: /eth2/b62f2b0e/attester_slashing/ssz_snappy
2025-07-08 13:19:35.841 INFO  - Subscribe to topic: /eth2/b62f2b0e/sync_committee_contribution_and_proof/ssz_snappy
2025-07-08 13:19:35.841 INFO  - Subscribe to topic: /eth2/b62f2b0e/bls_to_execution_change/ssz_snappy
2025-07-08 13:19:35.841 INFO  - Subscribe to topic: /eth2/b62f2b0e/data_column_sidecar_0/ssz_snappy
2025-07-08 13:19:35.841 INFO  - Subscribe to topic: /eth2/b62f2b0e/data_column_sidecar_33/ssz_snappy
2025-07-08 13:19:35.841 INFO  - Subscribe to topic: /eth2/b62f2b0e/data_column_sidecar_116/ssz_snappy
2025-07-08 13:19:35.841 INFO  - Subscribe to topic: /eth2/b62f2b0e/data_column_sidecar_50/ssz_snappy
2025-07-08 13:19:35.841 INFO  - Subscribe to topic: /eth2/b62f2b0e/data_column_sidecar_49/ssz_snappy
2025-07-08 13:19:35.841 INFO  - Subscribe to topic: /eth2/b62f2b0e/data_column_sidecar_100/ssz_snappy
2025-07-08 13:19:35.841 INFO  - Subscribe to topic: /eth2/b62f2b0e/data_column_sidecar_115/ssz_snappy
2025-07-08 13:19:35.841 INFO  - Subscribe to topic: /eth2/b62f2b0e/data_column_sidecar_67/ssz_snappy
2025-07-08 13:19:35.841 INFO  - Subscribe to topic: /eth2/b62f2b0e/data_column_sidecar_98/ssz_snappy
2025-07-08 13:19:35.841 INFO  - Subscribe to topic: /eth2/b62f2b0e/data_column_sidecar_65/ssz_snappy
....
2025-07-08 13:19:35.844 INFO  - Subscribe to topic: /eth2/b62f2b0e/data_column_sidecar_85/ssz_snappy
2025-07-08 13:19:35.845 INFO  - Subscribe to topic: /eth2/b62f2b0e/data_column_sidecar_87/ssz_snappy
2025-07-08 13:19:35.845 INFO  - Subscribe to topic: /eth2/b62f2b0e/data_column_sidecar_81/ssz_snappy
2025-07-08 13:19:35.845 INFO  - Subscribe to topic: /eth2/b62f2b0e/data_column_sidecar_99/ssz_snappy
2025-07-08 13:19:35.845 INFO  - Subscribe to topic: /eth2/b62f2b0e/data_column_sidecar_66/ssz_snappy
2025-07-08 13:19:36.329 INFO  - Subscribe to topic: /eth2/b62f2b0e/beacon_attestation_0/ssz_snappy
2025-07-08 13:19:36.332 INFO  - Subscribe to topic: /eth2/b62f2b0e/beacon_attestation_20/ssz_snappy
2025-07-08 13:19:36.332 INFO  - Subscribe to topic: /eth2/b62f2b0e/beacon_attestation_7/ssz_snappy
2025-07-08 13:19:36.333 INFO  - Subscribe to topic: /eth2/b62f2b0e/beacon_attestation_27/ssz_snappy
2025-07-08 13:19:36.333 INFO  - Subscribe to topic: /eth2/b62f2b0e/beacon_attestation_15/ssz_snappy
2025-07-08 13:19:36.511 ERROR - PLEASE FIX OR REPORT | Unexpected exception thrown for beaconchain-async-7
java.util.concurrent.CompletionException: java.util.NoSuchElementException: No value present
at java.base/java.util.concurrent.CompletableFuture.encodeThrowable(Unknown Source) ~[?:?]
at java.base/java.util.concurrent.CompletableFuture.completeThrowable(Unknown Source) ~[?:?]
at java.base/java.util.concurrent.CompletableFuture$UniAccept.tryFire(Unknown Source) ~[?:?]
at java.base/java.util.concurrent.CompletableFuture.postComplete(Unknown Source) ~[?:?]
at java.base/java.util.concurrent.CompletableFuture.completeExceptionally(Unknown Source) ~[?:?]
at tech.pegasys.teku.infrastructure.async.stream.AsyncIteratorCollector.onError(AsyncIteratorCollector.java:44) ~[teku-infrastructure-async-develop.jar:25.6.0+30-gaa18f64bdd]
at tech.pegasys.teku.infrastructure.async.stream.MapStreamHandler.onNext(MapStreamHandler.java:33) ~[teku-infrastructure-async-develop.jar:25.6.0+30-gaa18f64bdd]
at tech.pegasys.teku.infrastructure.async.stream.FlattenStreamHandler$1.onNext(FlattenStreamHandler.java:32) ~[teku-infrastructure-async-develop.jar:25.6.0+30-gaa18f64bdd]
at tech.pegasys.teku.infrastructure.async.stream.FutureAsyncIteratorImpl.lambda$iterate$1(FutureAsyncIteratorImpl.java:30) ~[teku-infrastructure-async-develop.jar:25.6.0+30-gaa18f64bdd]
at tech.pegasys.teku.infrastructure.async.SafeFuture.lambda$finish$33(SafeFuture.java:396) ~[teku-infrastructure-async-develop.jar:25.6.0+30-gaa18f64bdd]
at java.base/java.util.concurrent.CompletableFuture.uniHandle(Unknown Source) ~[?:?]
at java.base/java.util.concurrent.CompletableFuture$UniHandle.tryFire(Unknown Source) ~[?:?]
at java.base/java.util.concurrent.CompletableFuture.postComplete(Unknown Source) ~[?:?]
at java.base/java.util.concurrent.CompletableFuture.complete(Unknown Source) ~[?:?]
at tech.pegasys.teku.infrastructure.async.SafeFuture.lambda$propagateResult$3(SafeFuture.java:152) ~[teku-infrastructure-async-develop.jar:25.6.0+30-gaa18f64bdd]
at java.base/java.util.concurrent.CompletableFuture.uniWhenComplete(Unknown Source) ~[?:?]
at java.base/java.util.concurrent.CompletableFuture$UniWhenComplete.tryFire(Unknown Source) ~[?:?]
at java.base/java.util.concurrent.CompletableFuture.postComplete(Unknown Source) ~[?:?]
at java.base/java.util.concurrent.CompletableFuture.complete(Unknown Source) ~[?:?]
at tech.pegasys.teku.infrastructure.async.SafeFuture.lambda$propagateToAsync$24(SafeFuture.java:357) ~[teku-infrastructure-async-develop.jar:25.6.0+30-gaa18f64bdd]
at tech.pegasys.teku.infrastructure.async.SafeFuture.of(SafeFuture.java:84) ~[teku-infrastructure-async-develop.jar:25.6.0+30-gaa18f64bdd]
at tech.pegasys.teku.infrastructure.async.AsyncRunner.lambda$runAsync$2(AsyncRunner.java:47) ~[teku-infrastructure-async-develop.jar:25.6.0+30-gaa18f64bdd]
at tech.pegasys.teku.infrastructure.async.SafeFuture.of(SafeFuture.java:76) ~[teku-infrastructure-async-develop.jar:25.6.0+30-gaa18f64bdd]
at tech.pegasys.teku.infrastructure.async.ScheduledExecutorAsyncRunner.lambda$createRunnableForAction$1(ScheduledExecutorAsyncRunner.java:124) ~[teku-infrastructure-async-develop.jar:25.6.0+30-gaa18f64bdd]
at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source) ~[?:?]
at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source) ~[?:?]
at java.base/java.lang.Thread.run(Unknown Source) [?:?]
Caused by: java.util.NoSuchElementException: No value present
at java.base/java.util.Optional.get(Unknown Source) ~[?:?]
at tech.pegasys.teku.infrastructure.async.stream.MapStreamHandler.onNext(MapStreamHandler.java:31) ~[teku-infrastructure-async-develop.jar:25.6.0+30-gaa18f64bdd]
... 20 more
2025-07-08 13:19:36.696 INFO  - Subscribe to topic: /eth2/b62f2b0e/sync_committee_1/ssz_snappy
2025-07-08 13:19:36.702 INFO  - Subscribe to topic: /eth2/b62f2b0e/sync_committee_3/ssz_snappy
2025-07-08 13:19:37.381 ERROR - PLEASE FIX OR REPORT | Unexpected exception thrown for beaconchain-async-4
java.util.concurrent.CompletionException: java.util.NoSuchElementException: No value present
at java.base/java.util.concurrent.CompletableFuture.encodeThrowable(Unknown Source) ~[?:?]
at java.base/java.util.concurrent.CompletableFuture.completeThrowable(Unknown Source) ~[?:?]
at java.base/java.util.concurrent.CompletableFuture$UniAccept.tryFire(Unknown Source) ~[?:?]
at java.base/java.util.concurrent.CompletableFuture.postComplete(Unknown Source) ~[?:?]
at java.base/java.util.concurrent.CompletableFuture.completeExceptionally(Unknown Source) ~[?:?]
at tech.pegasys.teku.infrastructure.async.stream.AsyncIteratorCollector.onError(AsyncIteratorCollector.java:44) ~[teku-infrastructure-async-develop.jar:25.6.0+30-gaa18f64bdd]
at tech.pegasys.teku.infrastructure.async.stream.MapStreamHandler.onNext(MapStreamHandler.java:33) ~[teku-infrastructure-async-develop.jar:25.6.0+30-gaa18f64bdd]
at tech.pegasys.teku.infrastructure.async.stream.FlattenStreamHandler$1.onNext(FlattenStreamHandler.java:32) ~[teku-infrastructure-async-develop.jar:25.6.0+30-gaa18f64bdd]
at tech.pegasys.teku.infrastructure.async.stream.FutureAsyncIteratorImpl.lambda$iterate$1(FutureAsyncIteratorImpl.java:30) ~[teku-infrastructure-async-develop.jar:25.6.0+30-gaa18f64bdd]
at tech.pegasys.teku.infrastructure.async.SafeFuture.lambda$finish$33(SafeFuture.java:396) ~[teku-infrastructure-async-develop.jar:25.6.0+30-gaa18f64bdd]
at java.base/java.util.concurrent.CompletableFuture.uniHandle(Unknown Source) ~[?:?]
at java.base/java.util.concurrent.CompletableFuture$UniHandle.tryFire(Unknown Source) ~[?:?]
at java.base/java.util.concurrent.CompletableFuture.postComplete(Unknown Source) ~[?:?]
at java.base/java.util.concurrent.CompletableFuture.complete(Unknown Source) ~[?:?]
at tech.pegasys.teku.infrastructure.async.SafeFuture.lambda$propagateResult$3(SafeFuture.java:152) ~[teku-infrastructure-async-develop.jar:25.6.0+30-gaa18f64bdd]
at java.base/java.util.concurrent.CompletableFuture.uniWhenComplete(Unknown Source) ~[?:?]
at java.base/java.util.concurrent.CompletableFuture$UniWhenComplete.tryFire(Unknown Source) ~[?:?]
at java.base/java.util.concurrent.CompletableFuture.postComplete(Unknown Source) ~[?:?]
at java.base/java.util.concurrent.CompletableFuture.complete(Unknown Source) ~[?:?]
at tech.pegasys.teku.infrastructure.async.SafeFuture.lambda$propagateToAsync$24(SafeFuture.java:357) ~[teku-infrastructure-async-develop.jar:25.6.0+30-gaa18f64bdd]
at tech.pegasys.teku.infrastructure.async.SafeFuture.of(SafeFuture.java:84) ~[teku-infrastructure-async-develop.jar:25.6.0+30-gaa18f64bdd]
at tech.pegasys.teku.infrastructure.async.AsyncRunner.lambda$runAsync$2(AsyncRunner.java:47) ~[teku-infrastructure-async-develop.jar:25.6.0+30-gaa18f64bdd]
at tech.pegasys.teku.infrastructure.async.SafeFuture.of(SafeFuture.java:76) ~[teku-infrastructure-async-develop.jar:25.6.0+30-gaa18f64bdd]
at tech.pegasys.teku.infrastructure.async.ScheduledExecutorAsyncRunner.lambda$createRunnableForAction$1(ScheduledExecutorAsyncRunner.java:124) ~[teku-infrastructure-async-develop.jar:25.6.0+30-gaa18f64bdd]
at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source) ~[?:?]
at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source) ~[?:?]
at java.base/java.lang.Thread.run(Unknown Source) [?:?]
Caused by: java.util.NoSuchElementException: No value present
at java.base/java.util.Optional.get(Unknown Source) ~[?:?]
at tech.pegasys.teku.infrastructure.async.stream.MapStreamHandler.onNext(MapStreamHandler.java:31) ~[teku-infrastructure-async-develop.jar:25.6.0+30-gaa18f64bdd]
... 20 more
```

--- a/ethereum/statetransition/src/main/java/tech/pegasys/teku/statetransition/datacolumns/DataColumnSidecarRecoveringCustodyImpl.java
+++ b/ethereum/statetransition/src/main/java/tech/pegasys/teku/statetransition/datacolumns/DataColumnSidecarRecoveringCustodyImpl.java
@@ -235,6 +235,7 @@ public class DataColumnSidecarRecoveringCustodyImpl implements DataColumnSidecar
     final SafeFuture<List<DataColumnSidecar>> list =
         AsyncStream.createUnsafe(task.existingColumnIds().iterator())
             .mapAsync(delegate::getCustodyDataColumnSidecar)
+            .filter(Optional::isPresent)
             .map(Optional::get)
             .toList();
     initiateRecovery(task.block().get(), list);

